kotlin-native/litmus/src/commonMain/kotlin/SpinBarrier.kt

```diff
@@ -0,0 +1,16 @@
+import kotlin.native.concurrent.AtomicInt
+
+class SpinBarrier(private val threadCount: Int) {
+    private val waitingCount = AtomicInt(0)
+    private val passedBarriersCount = AtomicInt(0)
+
+    fun wait() {
+        val oldPassed = passedBarriersCount.value
+        if (waitingCount.addAndGet(1) == threadCount) {
```

Comment:
I assume operations on `AtomicInt` from `kotlin.native.concurrent` have `volatile` (sequentially consistent semantics). We need to re-check it in the code or ask developers. 

If that is true, then this code may result in additional heavyweight synchronization, that can interfere with testing results. For counter here we need relaxed (or `opaque` in Java terms) semantics here. That would have the minimal impact on testing results.